### PR TITLE
Release 4.2.0 into trunk

### PR DESCRIPTION
**Pull request by jkmassel**
New version 4.2.0. Be sure to create a GitHub Release and tag once this PR gets merged.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,20 @@ _None_
...
+
+_None_
+
+### Internal Changes
+
+_None_
+
+## 4.2.0
+
+### New Features
```
Comment (AliSoftware) [inline CHANGELOG.md:25]: Nit: this section contains both a new entry… and still kept the `_None_` line as well (that we were supposed to have removed when we added that first entry for #360 back then).

So you might as well take the occasion of this release PR to delete line 30 (sadly GitHub doesn't allow me to comment on line 30 to make a diff suggestion because that line is not part of this PR's diff 😝)

**Review by AliSoftware — APPROVED** [review]
Left a nit about the CHANGELOG otherwise LGTM :shipit:

File: CHANGELOG.md (hunk 1 above)
Comment (jkmassel) [reply CHANGELOG.md:25]: Nice and easy – addressed by https://github.com/wordpress-mobile/release-toolkit/pull/368/commits/d1e176aae92161376bb025f0659d29da37f1dbcf
